posts/2018-02-09-serverless-and-graphql-in-love.md

```diff
@@ -0,0 +1,608 @@
+
+It takes only a few minutes to get to a production-ready setup, which we're about to dive into, right now.
+
+## Serverless-GraphQL repository
+
+With the shiny new [Serverless and GraphQL Repository](https://github.com/serverless/serverless-graphql), it’s incredibly straightforward to get your HTTP endpoint up and running.
+
+![alt text](https://user-images.githubusercontent.com/1587005/36035218-1c06763c-0d6b-11e8-996b-996243b0975f.png "Serverless and GraphQL Architecture")
+
+The repository comes in two flavors: API Gateway + Lambda backend, or AppSync backend. (And we're currently working on adding more backend integrations, including [Graphcool Prisma](https://github.com/graphcool/prisma), Druid, MongoDB, and AWS Neptune.)
+
+**Note:** I’m going to focus on AWS Lambda below, but know that you can use any serverless provider (Microsoft Azure, Google Cloud Functions, etc) with GraphQL.
+
+## Let's create a Serverless GraphQL Endpoint
+
+In this example, we're using [`Apollo-Server-Lambda`](https://www.npmjs.com/package/apollo-server-lambda), designed to quickly get you up and running on AWS Lambda. Now you might be more familiar with server frameworks such as `Express`, `Koa`, or `Hapi`, and there are Apollo Server plugins for all of these frameworks. You can use any of these with Serverless as well, but they each require a little more configuration and adds complexity to the entire setup.
+Also, for monitoring your endpoint, you can integrate the lambda function with `Cloudwatch-metrics`, `AWS X-Ray` or [`Apollo Engine`](https://www.apollographql.com/engine/).
+
+Some of the main components of building your endpoint are:
+
+1. [handler.js](https://github.com/serverless/serverless-graphql/blob/master/app-backend/dynamodb/handler.js#L14): lambda function handler to route HTTP requests and return the response.
+
+2. [serverless.yml](https://github.com/serverless/serverless-graphql/blob/master/app-backend/dynamodb/serverless.yml#L34): creates AWS resources and sets up the GraphQL endpoint.
+
+3. [schema.js](https://github.com/serverless/serverless-graphql/blob/master/app-backend/dynamodb/schema.js): defines our GraphQL schema we're using to build this mini Twitter app.
+
+4. [resolver.js](https://github.com/serverless/serverless-graphql/blob/master/app-backend/dynamodb/resolvers.js): this is where we'll write query handler functions to fetch data from our other services (RDS, REST, DynamoDB, etc.)
+
+### Step 1: Configure Serverless template
+
+We'll be using the [Serverless Framework](https://serverless.com/framework/) to quickly build and deploy your API resources. You’ll specify in your `serverless.yml` that you are setting up a GraphQL HTTP endpoint.
+
+```javascript
+functions:
+  graphql:
+    handler: handler.graphqlHandler
+    events:
+    - http:
+        path: graphql
+        method: post
```

Comment:
`cors: true` is missing here

if we don't want cors in the example we can also simplify the code below updating the header

Reply:
ill add cors